### ISSUE TITLE
There's no way to search for two (or more) (key)words with AND as operator ???????

### ISSUE THREAD

**Issue by chrixxxx** [giuspen/cherrytree#752]
While using | and a regex I succeeded to make a search with 2 or more (key)words (the OR operator) .. I can't find a way to perform a search when 2 (or more) words that has to be present - ALL of them - in every document/node scanned by the search function.

Am I missing something ?

**Comment by txe** [comment]
(that.tools)|(tools.that)

**Comment by txe** [comment]
OK, it's not a good example, give a minute, I'll come up with something better

**Comment by txe** [comment]
that.*tools|tools.*that

**Comment by chrixxxx** [comment]
> that.*tools|tools.*that

Your syntax works ..but the search will remain limited to 2 key-words using that.

I think there should be a lot simpler - and more straightforward - way to search through notes/nodes, and that should be implemented at least the three main logical operators (OR, AND, NOT) when you make a search to make the search function a bit useful, especially when using tags.

**Comment by txe** [comment]
Right, or even simpler, to make search over the previous search, so you look up one word, then another, then some negative search.

**Comment by chrixxxx** [comment]
> Right, or even simpler, to make search over the previous search, so you look up one word, then another, then some negative search.

Yes.. that would be also a great solution, indeed. A fast and straightforward way to "narrowing" the results of a search applying new criteria only to such results .. until you have exactly what you was looking for.

**Comment by mahdi1234** [comment]
> Right, or even simpler, to make search over the previous search, so you look up one word, then another, then some negative search.

I would love this one, I'm using for years "that.*tools|tools.*that" but that's really limited only to two tags.

**Comment by JackDinn** [comment]
Was there never any movement on this? Its a PITA that every time i want to search for a node entry that i know should contain x, y & z somewhere within it, i have to go and re-learn (remember) how to use regex for such things again.

**Comment by gitvectors** [comment]
Although I have used CherryTree for a long time as a key part of my complex workflow  I still cannot find in Preferences how to captuse the filepath to the open CherryTree document although I see filerpath in top  toolbar. tere should be a command: "paste full filepath to this docume/node".

If I can copy this filepath (CherryTree command) I can create an internal CodeBox script to search though the document or indeed other CherryTree documents (many) scattered through my desktop. i use RipGrep to search for terms within *.ctd.
